packages/core/framework/src/http/utils/unless-path.ts

```diff
@@ -0,0 +1,17 @@
+import { MedusaNextFunction, MedusaRequest, MedusaResponse, MiddlewareFunction } from "../types"
+
+/**
+ * Due to how our route loader works, where we load all middlewares before routes, ambiguous routes * end up having all middlewares on different routes executed before the route handler is.
+ */
+/**
+ * This function allows us to skip middlewares for particular routes, so we can temporarily solve * * this without completely breaking the route loader for everyone.
+ */
+export const unlessPath =
+  (onPath: RegExp, middleware: MiddlewareFunction) =>
+  (req: MedusaRequest, res: MedusaResponse, next: MedusaNextFunction) => {
```

Comment:
Thought: Can you make all middleware named instead of anonymous to improve telemetry and stack trace please :D